{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this session, we are going to make practical use of what we just learned to explore a region of the genome, the RHD locus, which is responsible for the Rh antigen that can have an adverse effect during some pregnancies. This gene is located at chr1:25598977-25656936 in the hg19 version of the human genome reference sequence."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As with yesterday, set up imported packages and specific file locations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import re\n",
    "from pylab import plot,show\n",
    "from scipy import stats\n",
    "import os\n",
    "\n",
    "baseDir='/scratch/biobootcamp_fluxod/remills/day4/read_counts_by_region'\n",
    "\n",
    "fnSampleInfo = '%s/%s'%( baseDir, 'sample_info.txt' )\n",
    "fnSampleList = '%s/%s'%( baseDir, 'sample_names.txt' )\n",
    "\n",
    "#define region of interest\n",
    "chromo = \"1\"\n",
    "startPos = 25598977\n",
    "endPos = 25656936\n",
    "region = \"1:25598977-25656936\" #format needed for calling samtools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's take a quick look at the sample info file to check it's format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Sample',\n",
       " 'Family ID',\n",
       " 'Population',\n",
       " 'Population Description',\n",
       " 'Gender',\n",
       " 'File Path']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fileSampleInfo = open(fnSampleInfo,'r')\n",
    "line = fileSampleInfo.readline()\n",
    "fileSampleInfo.close()\n",
    "\n",
    "line = line.rstrip()\n",
    "line = line.split('\\t')\n",
    "line"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that this file is different from the one we used yesterday. It now includes an additional column, 'File Path', which gives an absolute location to the alignment file for each respective sample. As with yesterday, let's define a function to read through the file line by line and store the data for each sample and its file location into a dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def parseSampleTable( filename ):\n",
    "    fileSampTable = open(filename,'r')\n",
    "    \n",
    "    # skip past the file header\n",
    "    fileSampTable.readline()\n",
    "\n",
    "    samplePath={}\n",
    "    \n",
    "    # go line by line\n",
    "    for line in fileSampTable:\n",
    "        line = line.rstrip().split('\\t')\n",
    "        samplePath[ line[0] ] = line[5]\n",
    "    \n",
    "    return samplePath"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note how this function is almost identical to what we used yesterday. Reusing code is not only efficient but is recommended!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "samplePath = parseSampleTable( fnSampleInfo )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/scratch/data/1000genome/1000genomes/HG00112/alignment/HG00112.mapped.ILLUMINA.bwa.GBR.low_coverage.20120522.bam'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#example\n",
    "sample = \"HG00112\"\n",
    "samplePath[sample]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Our goal now is to count the sequence reads at the RHD locus from each file and compare them.  We can do this using the *samtools* command we learned yesterday combined with the unix command *wc -l* which will count the number of lines printed to thes screen. We can call this program from within python using the *subprocess.call()* function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "samtools view /scratch/data/1000genome/1000genomes/HG00112/alignment/HG00112.mapped.ILLUMINA.bwa.GBR.low_coverage.20120522.bam 1:25598977-25656936 | wc -l > HG00112.cnt\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#construct command as if we would be running it from the command line. \n",
    "command = \"samtools view \" + samplePath[sample] + \" \" + region + \" | wc -l > \" + sample + \".cnt\"\n",
    "print command\n",
    "subprocess.call(command, shell=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Each individual genome may be sequenced to different depths of coverage, leading to differences in read counts being a reflection of that depth rather than a true difference in sequence content. Thus, it would make sense to normalize by the total number of reads generated either for the whole genome or, as we will do here, for the chromosome of interest."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "samtools idxstats /scratch/data/1000genome/1000genomes/HG00112/alignment/HG00112.mapped.ILLUMINA.bwa.GBR.low_coverage.20120522.bam | head -1 |cut -f 3 > HG00112.num\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Only need data for chromosome 1, so 'head -1' returns only the first line of the output that corresponds to chromo 1\n",
    "#Number of aligned reads is in column 3, so we can 'cut' this column out and be left with only a single data point\n",
    "command = \"samtools idxstats \" + samplePath[sample] + \" | head -1 |cut -f 3 > \" + sample + \".num\"\n",
    "print command\n",
    "subprocess.call(command, shell=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "However, with over 2500 sequences it would be inefficient (and boring!) to do all of them manually in this fashion. We could make use of cluster computing to run all 2500 simultaneously, however this would result in 2500 individual jobs that would need to be submitted, managed, and reviewed.\n",
    "\n",
    "Instead, we can *batch* samples into smaller collections and send each batch as its own job. To do this, we will need to determine both **(a)** the number of batches we wish to submit and **(b)** the number of samples to include in each batch. For this exercise, we will construct batches of n=50."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2535\n",
      "51\n"
     ]
    }
   ],
   "source": [
    "numBatches = 50\n",
    "#length of dictionary = number of samples here, can be calculated with len(dict)\n",
    "numSamplesPerBatch = int(len(samplePath) / numBatches) + 1; #add one to round up if needed\n",
    "\n",
    "#create a directory to store all the files\n",
    "if not os.path.exists(baseDir + \"/counts\"):\n",
    "    os.makedirs(baseDir + \"/counts\")\n",
    "\n",
    "#create a directory to store PBS files\n",
    "if not os.path.exists(baseDir + \"/jobs\"):\n",
    "    os.makedirs(baseDir + \"/jobs\") \n",
    "\n",
    "#create a directory to store stdout and stderr files\n",
    "if not os.path.exists(baseDir + \"/logs\"):\n",
    "    os.makedirs(baseDir + \"/logs\") \n",
    "\n",
    "print len(samplePath)\n",
    "print numSamplesPerBatch"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then, we can use python to create individual PBS files for each batch and subsequently submit them to the cluster."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "lastSampleNum = 0;\n",
    "samples = samplePath.keys()\n",
    "\n",
    "for batchNum in range (numBatches): #for each batch we plan to run\n",
    "\n",
    "    #create new PBS file for submission, with corresponding header\n",
    "    fileBatch = open(baseDir+\"/jobs/batch\"+str(batchNum)+\".pbs\", \"w\")\n",
    "    fileBatch.write(\"#!/bin/bash\\n\") #shell line\n",
    "    fileBatch.write(\"#\\n\")\n",
    "    fileBatch.write(\"#PBS -N batch\" + str(batchNum) + \"\\n\") #name of job\n",
    "    fileBatch.write(\"#PBS -o \" + baseDir + \"/logs/batch\" + str(batchNum) + \".stdout\\n\") #stdout file\n",
    "    fileBatch.write(\"#PBS -e \" + baseDir + \"/logs/batch\" + str(batchNum) + \".stderr\\n\") #stderr file\n",
    "    fileBatch.write(\"#PBS -l procs=1,qos=flux,mem=4gb,walltime=24:00:00\\n\") #feature line\n",
    "    fileBatch.write(\"#PBS -m a\\n\") #message line, only send errors to email\n",
    "    fileBatch.write(\"#PBS -M remills@umich.edu\\n\") #email line, to send messages from above\n",
    "    fileBatch.write(\"#PBS -A biobootcamp_fluxod\\n\") #which account to associate job\n",
    "    fileBatch.write(\"#PBS -q fluxod\\n\") #which queue to send job to \n",
    "    fileBatch.write(\"#PBS -V\\n\") #pass environmental variables to job\n",
    "    fileBatch.write(\"#PBS -d .\\n\") #use current working directory\n",
    "    fileBatch.write(\"\\n\") #use current working directory\n",
    "    \n",
    "    startIndex = lastSampleNum\n",
    "    endIndex = lastSampleNum + numSamplesPerBatch\n",
    "    \n",
    "    #make sure last index is in range, in case batches aren't divided evenly\n",
    "    if endIndex > len(samplePath): \n",
    "        endIndex = len(samplePath)\n",
    "        \n",
    "    for sampleNum in range (startIndex, endIndex):\n",
    "        sample = samples[sampleNum] #current sample\n",
    "        #write command for region counts\n",
    "        command = \"samtools view \" + samplePath[sample] + \" \" + region + \" | wc -l > \" + baseDir + \"/counts/\" + sample + \".cnt\"\n",
    "        fileBatch.write(command+\"\\n\")\n",
    "        #write command for chromosome counts\n",
    "        command = \"samtools idxstats \" + samplePath[sample] + \" | head -1 | cut -f 3 > \" + baseDir + \"/counts/\" + sample + \".num\"\n",
    "        fileBatch.write(command+\"\\n\")\n",
    "    fileBatch.close()\n",
    "    \n",
    "    #submit job to cluster\n",
    "    #subprocess.call([\"qsub\", baseDir+\"/jobs/batch\"+str(batchNum)+\".pbs\"]) \n",
    "    \n",
    "    #update index for next iteration\n",
    "    lastSampleNum = lastSampleNum + numSamplesPerBatch\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we must wait for the jobs to finish. There are python modules which will check and sleep until all jobs are finalized. For simplicity, we can just manually check with the *qstat* command you were introduced to earlier (use your unique name instead of mine!)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "nyx.arc-ts.umich.edu: \r\n",
      "                                                                                  Req'd    Req'd       Elap\r\n",
      "Job ID                  Username    Queue    Jobname          SessID  NDS   TSK   Memory   Time    S   Time\r\n",
      "----------------------- ----------- -------- ---------------- ------ ----- ------ ------ --------- - ---------\r\n",
      "16694971.nyx.arc-ts.um  remills     fluxod   batch0             9026   --       1  768mb  24:00:00 C       -- \r\n",
      "16694983.nyx.arc-ts.um  remills     fluxod   batch0             9500   --       1  768mb  24:00:00 C       -- \r\n",
      "16694993.nyx.arc-ts.um  remills     fluxod   batch1             9838   --       1  768mb  24:00:00 C       -- \r\n",
      "16695026.nyx.arc-ts.um  remills     fluxod   batch1            10286   --       1  768mb  24:00:00 C       -- \r\n",
      "16695046.nyx.arc-ts.um  remills     fluxod   batch0            10649   --       1  768mb  24:00:00 C       -- \r\n",
      "16695047.nyx.arc-ts.um  remills     fluxod   batch0            11065   --       1  768mb  24:00:00 C       -- \r\n",
      "16695057.nyx.arc-ts.um  remills     fluxod   batch0            11386   --       1  768mb  24:00:00 C       -- \r\n",
      "16695058.nyx.arc-ts.um  remills     fluxod   batch1             9771   --       1  768mb  24:00:00 C       -- \r\n",
      "16695059.nyx.arc-ts.um  remills     fluxod   batch2             9777   --       1  768mb  24:00:00 C       -- \r\n",
      "16695060.nyx.arc-ts.um  remills     fluxod   batch3            17546   --       1  768mb  24:00:00 C       -- \r\n",
      "16695061.nyx.arc-ts.um  remills     fluxod   batch4            17552   --       1  768mb  24:00:00 C       -- \r\n",
      "16695062.nyx.arc-ts.um  remills     fluxod   batch5            30238   --       1  768mb  24:00:00 C       -- \r\n",
      "16695063.nyx.arc-ts.um  remills     fluxod   batch6            30243   --       1  768mb  24:00:00 C       -- \r\n",
      "16695064.nyx.arc-ts.um  remills     fluxod   batch7             9291   --       1  768mb  24:00:00 C       -- \r\n",
      "16695065.nyx.arc-ts.um  remills     fluxod   batch8             9296   --       1  768mb  24:00:00 C       -- \r\n",
      "16695066.nyx.arc-ts.um  remills     fluxod   batch9            32503   --       1  768mb  24:00:00 C       -- \r\n",
      "16695067.nyx.arc-ts.um  remills     fluxod   batch10           32507   --       1  768mb  24:00:00 C       -- \r\n",
      "16695068.nyx.arc-ts.um  remills     fluxod   batch11           26182   --       1  768mb  24:00:00 C       -- \r\n",
      "16695069.nyx.arc-ts.um  remills     fluxod   batch12           26187   --       1  768mb  24:00:00 C       -- \r\n",
      "16695070.nyx.arc-ts.um  remills     fluxod   batch13           32616   --       1  768mb  24:00:00 C       -- \r\n",
      "16695071.nyx.arc-ts.um  remills     fluxod   batch14           32621   --       1  768mb  24:00:00 C       -- \r\n",
      "16695072.nyx.arc-ts.um  remills     fluxod   batch15           17023   --       1  768mb  24:00:00 C       -- \r\n",
      "16695073.nyx.arc-ts.um  remills     fluxod   batch16           17027   --       1  768mb  24:00:00 C       -- \r\n",
      "16695074.nyx.arc-ts.um  remills     fluxod   batch17            8517   --       1  768mb  24:00:00 C       -- \r\n",
      "16695075.nyx.arc-ts.um  remills     fluxod   batch18            8522   --       1  768mb  24:00:00 C       -- \r\n",
      "16695076.nyx.arc-ts.um  remills     fluxod   batch19           15911   --       1  768mb  24:00:00 C       -- \r\n",
      "16695077.nyx.arc-ts.um  remills     fluxod   batch20           11683   --       1  768mb  24:00:00 C       -- \r\n",
      "16695078.nyx.arc-ts.um  remills     fluxod   batch21           10360   --       1  768mb  24:00:00 C       -- \r\n",
      "16695079.nyx.arc-ts.um  remills     fluxod   batch22           10380   --       1  768mb  24:00:00 C       -- \r\n",
      "16695080.nyx.arc-ts.um  remills     fluxod   batch23           18136   --       1  768mb  24:00:00 C       -- \r\n",
      "16695081.nyx.arc-ts.um  remills     fluxod   batch24           18156   --       1  768mb  24:00:00 C       -- \r\n",
      "16695082.nyx.arc-ts.um  remills     fluxod   batch25           30828   --       1  768mb  24:00:00 C       -- \r\n",
      "16695083.nyx.arc-ts.um  remills     fluxod   batch26           30848   --       1  768mb  24:00:00 C       -- \r\n",
      "16695084.nyx.arc-ts.um  remills     fluxod   batch27            9976   --       1  768mb  24:00:00 C       -- \r\n",
      "16695085.nyx.arc-ts.um  remills     fluxod   batch28            9996   --       1  768mb  24:00:00 C       -- \r\n",
      "16695086.nyx.arc-ts.um  remills     fluxod   batch29             636   --       1  768mb  24:00:00 C       -- \r\n",
      "16695087.nyx.arc-ts.um  remills     fluxod   batch30             656   --       1  768mb  24:00:00 C       -- \r\n",
      "16695088.nyx.arc-ts.um  remills     fluxod   batch31           26772   --       1  768mb  24:00:00 C       -- \r\n",
      "16695089.nyx.arc-ts.um  remills     fluxod   batch32           26792   --       1  768mb  24:00:00 C       -- \r\n",
      "16695090.nyx.arc-ts.um  remills     fluxod   batch33             751   --       1  768mb  24:00:00 C       -- \r\n",
      "16695091.nyx.arc-ts.um  remills     fluxod   batch34             771   --       1  768mb  24:00:00 C       -- \r\n",
      "16695092.nyx.arc-ts.um  remills     fluxod   batch35           17615   --       1  768mb  24:00:00 C       -- \r\n",
      "16695093.nyx.arc-ts.um  remills     fluxod   batch36           17635   --       1  768mb  24:00:00 C       -- \r\n",
      "16695094.nyx.arc-ts.um  remills     fluxod   batch37            9123   --       1  768mb  24:00:00 C       -- \r\n",
      "16695095.nyx.arc-ts.um  remills     fluxod   batch38            9143   --       1  768mb  24:00:00 C       -- \r\n",
      "16695096.nyx.arc-ts.um  remills     fluxod   batch39           16218   --       1  768mb  24:00:00 C       -- \r\n",
      "16695097.nyx.arc-ts.um  remills     fluxod   batch40           12085   --       1  768mb  24:00:00 C       -- \r\n",
      "16695098.nyx.arc-ts.um  remills     fluxod   batch41           10946   --       1  768mb  24:00:00 C       -- \r\n",
      "16695099.nyx.arc-ts.um  remills     fluxod   batch42           10960   --       1  768mb  24:00:00 C       -- \r\n",
      "16695100.nyx.arc-ts.um  remills     fluxod   batch43           18818   --       1  768mb  24:00:00 C       -- \r\n",
      "16695101.nyx.arc-ts.um  remills     fluxod   batch44           18838   --       1  768mb  24:00:00 C       -- \r\n",
      "16695102.nyx.arc-ts.um  remills     fluxod   batch45           31414   --       1  768mb  24:00:00 C       -- \r\n",
      "16695103.nyx.arc-ts.um  remills     fluxod   batch46           31434   --       1  768mb  24:00:00 C       -- \r\n",
      "16695104.nyx.arc-ts.um  remills     fluxod   batch47           10562   --       1  768mb  24:00:00 C       -- \r\n",
      "16695105.nyx.arc-ts.um  remills     fluxod   batch48           10582   --       1  768mb  24:00:00 C       -- \r\n",
      "16695106.nyx.arc-ts.um  remills     fluxod   batch49            1269   --       1  768mb  24:00:00 C       -- \r\n"
     ]
    }
   ],
   "source": [
    "!qstat -u remills"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This might take a few minutes to run. Once completed, we should take the time to do some quick checks of the stderr files to make sure that there were not  issues with any particular batch. This can be as simple as a quick *cat* of all the stderr files to see if anything is there. The lack of errors sent to your email address is also a sign that things went well."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n",
      "Note well:\n",
      "\n",
      "loading openmpi/1.6.0/gcc/4.4.6 and gcc/4.4.6 for Rmpi/snow\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!cat {baseDir}/logs/*stderr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we can use python to read back in all the output files and construct a normalized list of sequence counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "samples = samplePath.keys()\n",
    "rhdList = []\n",
    "for sample in samples:\n",
    "    #rhd counts\n",
    "    cntFile = open(baseDir + \"/counts/\" + sample + \".cnt\", \"r\")\n",
    "    cnt = cntFile.readline()\n",
    "    cnt = float(cnt.rstrip()) #float is important, otherwise will treat as STR or INT\n",
    "    cntFile.close()\n",
    "    \n",
    "    #chromo 1 counts\n",
    "    numFile = open(baseDir + \"/counts/\" + sample + \".num\", \"r\")\n",
    "    num = numFile.readline()\n",
    "    num = float(num.rstrip())\n",
    "    numFile.close()\n",
    "    \n",
    "    normCnt = 1e3 * (cnt / num) #scale for plotting\n",
    "    rhdList.append(normCnt)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYsAAAEZCAYAAABmTgnDAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAHxRJREFUeJzt3Xm4HFWd//H3hwBCWBIRDMjqhogDGDYjblcUBhEDig/o\nuKCizIw6zrgH+Y1eHx1HxF1HnREHIzqyKMMIbgSkFVfAhEUBETCyJiwCIkQI8P39UaeTonO7q/r2\nUtV9P6/nuc/trq7l26furW+dc6pOKSIwMzPrZL2qAzAzs/pzsjAzs0JOFmZmVsjJwszMCjlZmJlZ\nIScLMzMr5GRhNiIkfVXSh6qOw2YmJwvriqTlku6TdI+kFZJOlrR57vN1DmiSdpL0sKT1cvPcL+nP\n6edySR/Jr2eK7U5KOnlw32x6UlyrU3ncJemXkp4zoM1F+mkXyzaSviLp5lSuV6b4Zg8onuZ2a7lv\nrL+cLKxbARwSEZsBewC7Af+v5fOiOz0DOD4iNge2BF4PLAB+1uHAVte7RwP4ZiqPxwDnAt8a4PY0\n5URpC+AXwKOABalsDwDmAE8cYDw2QzhZ2LRFxErgHOBp01hcaR0PRMTFwEKyg+3rO80/5QfSQkm/\nlXSnpPMl7ZL7bHtJZ0i6VdLtkj6Xpj/ibHiK2s/rJF2bztCvk/R3HeJqfpeHgP8BtpK0ZVrPnNzZ\n/o2SPpTbxhMl/SjFdZukr0uak4tpvqSlKYZTgI06lOc7gLsj4tURcX2K58aIeHtEXJ7Wt5+ki1IN\n6EJJz8xta7mkF+TerymfXNm8VtIfU6zvS58dBBwLHJlqV8u6LD8bEU4WNh0CkLQdcBDwq6k+70ZE\n/AVYAnTVhCNpZ7ID9NvIainfA86StL6kWcDZwB+AHYFtgW82N9lhnZsAnwEOSmfozwQuKRHLhsBr\ngWsj4vY0+avAA2Rn9/OBA4E35hb7N2Ab4KnA9sBkbl1nAouBRwOnA4d3iPuFwBkdYtsC+C7waWAL\n4JPAdyU9Os3SWiOcajvPAnYGXgC8X9JTIuIHwEeAUyJis4iYP93ys3pzsrBuCThT0p+B64FrgQ+3\nfP6udJZ/p6Q7gUsp14x0C9mBrBtHAmdHxHnpzP7jwMZkB7Z9yQ7E746IVRFxf0T8PBdnJw8Du0na\nOCJWRsQVHeY9In3P+8gSwcEAkuYBLwLenrZ/G9nB+hUAEXFtint1Si6fAp6X1rkAWD8iPhMRD0XE\nt4GLOsSwBVn5tfNi4HcR8Y2IeDgiTgGuAl7SZv6pyueDqQwvI9une+TmbZ2/m/KzEeBkYd0K4NB0\nxjgB7A/s3fL5CRHx6OYPsDvlahvbAnd0Gc82ZEkr23g2MuYNaV3bAX+MiIe7WWFE3EuWhP4BuFnS\n2ZKe0mGRU9P3nAf8BvinNH1HYAPgllzi/BKwFWTJRNIpqXnqbuBksqY4gMcBN7Vs54+0L8c70jLt\nPI5cOeXWt22HZVqtyL2+D9h0qpmmUX42ApwsbNoi4ifA54DjWz5qPaBNdYB7RE1D0qZkTSkXtNtc\nm+k3kx2Um+sRWXPOjWRJY4fUHNXqL0C+M33rR2ws4pyIODBNvwr4coe4mn0WdwDHAMdIenza/v3A\nY3LJc05E7JaW/QjwEPA3ETEHeA1r/ydvYd0D+Y60L4dzgZem7z+Vm8iVU259zYR0L7BJ7rOtKW+d\nmLooPxsRThbWq08D+0p6RnpfpgaxptlC0qMk7UXWPn8HcFKHZdZL82+Ufh4FnAa8WNL+kjYA3gn8\nFfg5WbPNLcBHJc1Oy+yX1ncJ8NzUAT6HrJOWFNNjJR2a2t5Xkx1IH+oQ1xoRcTVwFvCeiLiF7AKA\nT0raTNJ6qVP7uWn2TdO6/yxpW+DduVX9AnhQ0tskbSDpZcA+7QqUrA9ic2CxpB3S99hW0ick7UbW\nl7OzpFem/pwjgV3I+nSa5fGK9NnedO4fabUC2KmZqLosPxsRThbWk9TWvhh4b3MSUx9kWjtP35P6\nPZrLXwTsFxGr2m0KeCWwiqwJ5D7g9+ng/GqyGs5tZG3zL4mIB1MfxkuAJ5E1wdwAHJHiPhc4Fbgs\nbfusXIzrAW8nO+u+g6zT/R87xNX6fU8AXivpsWQd3hsCVwB/Iuuobp61fxDYE7g7bf/bzXVFxAPA\ny4DXpRiOSJ9PHUTEncB+ZAfnX6WyPRe4C7gmIv4EHEKWTG8H3kV2CfSf0ir+lawT/k6yTvZvTPE9\n2zk9/b5D0sV0V342IjSohx9J+m+yf9xbm9XudEXGqWTV3+XAERFxV/rsWOANZGcgb4uIcwYSmJmZ\ndW2QNYuTyC6rzFsELImInYHz0nsk7UrWIbZrWuYLzWvRzcysegM7IEfEBWRV2ryFZE0OpN+HpdeH\nkt0FuzoilgPXkF32aGZmNTDss/d56a5fgJVklxpCdlnfjbn5bqS7S/rMzGyAKmvqSdfDd+owqetY\nQGZmM876Q97eSklbR8QKSdsAt6bpN5FdG9+0HevekIQkJxAzs2mIiK6H4ckbds3iO8BR6fVRZNfW\nN6e/QtKG6WamJwMXTrWCiKjdzwc+8IHKY3BMjmkmxuWYyv30w8BqFpK+STbOzZaSbgDeD3wUOE3S\n0aRLZwEi4gpJp5Fdi/4g8Obo1zc0M7OeDSxZRMQr23z0wjbzf4Rs+AMzM6sZ38vQBxMTE1WHsA7H\nVI5jKq+OcTmm4RnYHdyDIMmtU2ZmXZJEjFgHt5mZjSAnCzMzK+RkYWZmhYZ9U56ZDUnrc5Dc32e9\ncM3CbKwVjapjVo6ThZmZFXKyMDOzQk4WZmZWyMnCzMwKOVmYmVkhJwszMyvkZGFmZoWcLMzMrJCT\nhZmZFXKyMDOzQk4WZmZWyMnCzMwKOVmYmVkhD1FuA+MhsuvJ+8WmwzULGzAPkV1P3i/WHScLMzMr\n5GRhZmaFnCzMzKyQk4WZmRVysjAzs0JOFmZmVsjJwszMCvmmPLMx0nrDXbfL+AY9a8c1C7Ox0+0N\nd75Bz4o5WZiZWSEnCzMzK+RkYWZmhdzBbTYiPFqsVck1C7OR4s5oq0YlyULSsZJ+K+lySf8j6VGS\ntpC0RNLVks6RNLeK2MzGlaRpXVprBhUkC0k7AW8C9oyI3YBZwCuARcCSiNgZOC+9N7O+ca3Epq+K\nmsWfgdXAbEnrA7OBm4GFwOI0z2LgsApiM5vRmrUP10Cs1dCTRUT8CfgEcD1ZkrgrIpYA8yJiZZpt\nJTBv2LGZmWsfNrWhXw0l6YnAvwA7AXcDp0t6dX6eiAhJU/7FTk5Ornk9MTHBxMTEoEI1MxtJjUaD\nRqPR13Vq2JffSToSOCAi3pjevwZYAOwPPD8iVkjaBjg/InZpWTZ8ueDoyJoymvtLvtSzR2XKs3We\nXl57f40PSURET22LVfRZXAUskLSxsr/sFwJXAGcBR6V5jgLOrCA2MzObwtCboSLiUklfAy4GHgaW\nAv8FbAacJuloYDlwxLBjMzOzqQ29GaoXboYaLW6G6i83Q9l09aMZysN92NB52Ir+8HMobJg83IdV\nxJdo9s5laMPjZGFmZoWcLMzMrJCThZmZFXKyMDOzQk4WZmZWyMnCzMwKOVmYmVkhJwszMyvkZGFm\nZoU83IeZdeThWQxcszCzUjy0yEznZGFmZoWcLMzMrJD7LKxyHmrbrP5cs7AacHu4Wd05WZiZWSE3\nQ1lftV5maWbjwTULGwA3K5mNG9cszMaAa3Q2aK5ZmI0F1+ZssJwszMyskJOFmZkVcrIwM7NCThZm\nZlbIycLMzAo5WZiZWSEnCzMzK+RkYWZmhZwszMyskJOFmZkVcrIwM7NCHkjQrMY8QKDVhWsWZrXn\nQQKtepUkC0lzJX1L0pWSrpD0DElbSFoi6WpJ50iaW0VsZma2rqpqFp8BvhcRTwV2B64CFgFLImJn\n4Lz03szMakARw63eSpoDLIuIJ7RMvwp4XkSslLQ10IiIXVrmiWHHa93J2tib++iRr5v7rsw8lulU\nVoN+3W5/eR+NHklERE8dYIU1C0m79bKBKTweuE3SSZKWSvqypE2AeRGxMs2zEpjX5+1axSS5w9Zs\nRJVphvqipIskvTnVCnq1PrAn8IWI2BO4l5Ymp1R98OnL2PFuNRtVhZfORsSzJe0MvAFYKulC4KSI\nOGea27wRuDEiLkrvvwUcC6yQtHVErJC0DXDrVAtPTk6ueT0xMcHExMQ0wzAzG0+NRoNGo9HXdZbu\ns5C0PnAY8FngbrJayfsi4ttdb1T6CfDGiLha0iQwO310R0QcL2kRMDciFrUs5z6Lmuu1jd3795Hc\nZ2H90I8+i8JkIWkP4HXAIcAS4MSIWCrpccAvI2KHrjearfNEYEPgWuD1wCzgNGAHYDlwRETc1bKc\nk0XNOVn0bt1+HScL682wksWPga8A34qI+1o+e21EfK2XALrhZFF/Tha9q7I24WQxnoaVLDYFVkXE\nQ+n9LGCjiLi3lw1Ph5NF/TlZ9M7JwvptKJfOAucCG+fezyZrjjIzsxmiTLLYKCL+0nwTEfewtkPa\nzMxmgDLJ4l5JezXfSNobWDW4kMzMrG7KDFH+L8Bpkm5J77cBjhxcSGZmVjel7rOQtCHwFLJert9F\nxOpBB9YmDndw15w7uKenLpfLuoN7PPWjg7vsw4/2JhvTaX1gz7ThoV0yazYz5A/aZvVSmCwkfR14\nAnAJ8FDuIycLM7MZokzNYi9gV7f/mJnNXGWuhvoNWae2mZnNUGVqFlsBV6TRZu9P0yIiFg4uLDMz\nq5MyyWIy/Q7W9ry5ScrMbAYp8zyLhqSdgCdFxLmSZpdZzszMxkeZx6oeA5wO/GeatB3wv4MMyszM\n6qVMB/dbgGcDfwaIiKuBxw4yKJu5ms/p9rO6q+f9YHllksX9EdHs2G4+Mc99FjYgfk53fXhf2Fpl\nksWPJR0HzJZ0AFmT1FmDDcvM6i5fC3QNZPyVefjRLOBo4MA06Ydkj1Yd+imHx4aqv34+uGcm7eu6\nPPDIY0aNp6E8Ka9OnCzqaVCD4M2kfe1kYYM0lIEEJf1hiskREU/oZcM2bjwIntk4K3O/xD651xsB\nLwceM5hwzGaOUW3nH9W4rTfTaoaStDQi9hxAPEXbdTNUDQ2qCWXc9/UoNT15f422YTVD7cXav4j1\nyJ5tMauXjZqZ2Wgp0wz1CdYmiweB5cARgwrIzMzqp8zYUBNDiMPMzGqsTDPUO1n3Ns41o89GxCf7\nHpWZmdVK2Sfl7QN8hyxJHAJcBFw9wLjMzKxGytzBfQFwcETck95vBnwvIp4zhPhaY/HVUDXkq6Gm\nx1dD2bD042qoMmNDPRZYnXu/Go86a2Y2o5RphvoacKGkM8hOJQ4DFg80KjMzq5VSN+Wley2end7+\nJCKWDTSq9nG4GaqG3Aw1PW6GsmEZVjMUwGzgnoj4DHCjpMf3slEzMxstZR6rOgm8B1iUJm0IfH2A\nMQ2Vx+Q3MytWpmbxUuBQ4F6AiLgJ2GyQQQ1f4CeCmZm1V/axqg8330jaZIDxmJlZDZVJFqdL+k9g\nrqRjgPOAEwcblpmZ1UnHq6GUNeJvD+xC7rGqEbGk5w1nj2u9GLgxIl4iaQvgVGBH0mCFEXFXyzJ9\nvxrKT/vq3aCvhmrtSxq1fdQufl8NZcMy8MeqpmRxeUT8TS8babPud5ANJbJZRCyU9DHg9oj4mKT3\nAo+OiEUtyzhZ1NBwksXo7qN28TtZ2LAM/NLZdGT+taR9e9lIK0nbAQeTNWc1v8BC1t7st5js5j8z\nM6uBMndwLwBeLemPpCuiyPLI7j1s91PAu4HNc9PmRcTK9HolMK+H9ZuZWR+1TRaSdoiI64G/Jatr\n9uUmBEmHALdGxDJJE1PNExEhyXVaM7Oa6FSz+D9gfkQsl/TtiDi8T9vcD1go6WBgI2BzSScDKyVt\nHRErJG0D3DrVwpOTk2teT0xMMDEx0aewMvnOSLfBVmtcb5Ic1+9l9dFoNGg0Gn1dZ9sObknLImJ+\n6+u+blx6HvCudDXUx4A7IuJ4SYuAuVV0cLvDrnvD7qgdtf0ybh3Z47JfZpJhjg01SM2/sI8CB0i6\nGtg/vTczsxroVLN4CLgvvd0YWJX7OCJi83WXGizXLOrJNYvOXLOwqvWjZtG2zyIiZvWyYjMzGx91\naIYyM7Oac7IwM7NCThZmZlbIycLMzAo5WZiZWSEnCzMzK1RmIEEzK2HUn7th1olrFmZ9Ffh57jaO\nnCzMzKyQk4WZmRVysjAzs0JOFmZmVshXQ5lZX/jBYePNNQsz6xNfCTbOnCzMzKyQm6HMrO/cJDV+\nXLMwswFwk9S4cc3CzIbCw6GMNtcszGyIXOMYVU4WZmZWyMnCzMwKuc/CpqW1/bmqbde53bvKMjLr\nN9csrAdVtT+PSrv3qMRpVszJwszMCrkZyswGys1x48E1CzMbMDfHjQMnCzMzK+RkYWZmhZwszMys\nkJOFmZkV8tVQZlYZDy44OlyzMLOK+WqpUeCaRQejMqyEmdmguWbRkc94zMxgBtYsDj/8VVxyyZUA\nbLBBxcGYmY2IodcsJG0v6XxJv5X0G0lvS9O3kLRE0tWSzpE0dxDbv+yy33Hdde/guutO5KabVg1i\nE2NL0pofM5tZqmiGWg28PSKeBiwA3iLpqcAiYElE7Aycl94PyC7AnsyaNXtwmxhbbpozm4mGniwi\nYkVEXJJe/wW4EtgWWAgsTrMtBg4bdmxmZja1Sju4Je0EzAd+BcyLiJXpo5XAvIrCMjOzFpV1cEva\nFPg28M8RcU/LZaohacq2jsnJyTWvJyYmmJiYGGygZmYjptFo0Gg0+rpOVXH/gKQNgLOB70fEp9O0\nq4CJiFghaRvg/IjYpWW56DXeJz95b6655kvA3syZsxd3372UtW3wavva91k07zspLqthvq7Tfqlj\n+dT5dUSsU2Z12p/jRBIR0dOVKVVcDSXgK8AVzUSRfAc4Kr0+Cjhz2LHZ6PEVWmbDUUUz1LOAVwOX\nSVqWph0LfBQ4TdLRwHLgiApis5GTP2s1s0EZerKIiJ/SvkbzwmHGYmZm5Xi4DzMzK+RkYWZmhZws\nzMyskJOFmZkVcrIwM7NCThZmZlZoxj3PYrr81Dwzm8lcsyjNQ3Ob2czlZGFmZoXcDGUdecwlGyY3\n99aXaxZWgpvgbFj8t1ZXrlmY9cA1L5spXLMw65nPhm38OVmYmVkhJwszMyvkPgsbO639CL6qpp76\n2d/jfT54rlnYmHI/Qv31e/94nw+Saxa2Dl/hsy6fuQ6f77moF9csrA2fpa3LZTJcLu86cbIwM7NC\nboYyYLybntp9t2bTRrsmpnEuk1EzVZOU989wuWZhOeNa7c9/r3bfsdvpNlzeP1VzzcLGRr/ONH3G\nOp58kUJvXLOwMdKvs0yfrY4v79vpcs3CbBpc+6g3X3bbf65ZmE2Lz1Drzfun31yzsBnLtYPR5P1W\nDdcsbAbz2edo8n6rgpOFmZkVcrIwM7NCThZmZlbIHdxmNtbcId4frlmY2Zhzh3g/uGYxDeNyw4/P\nuMzK8VAhrllM0zidqYzTdzEbpJn9v1KrZCHpIElXSfq9pPdWHU8Zktb8jIJRi9dsUIr+Fzp9PhP/\nj2qTLCTNAj4PHATsCrxS0lOrjaqMAM6vOoh1NBqNDp9WdYbUqGCbRRpVBzBCGlUHMIVGD8uW+T/o\nfmj0zv97o6s2yQLYF7gmIpZHxGrgFODQimMqqVF1AOuo5x9so+oAptCoOoAR0qg6gCk0qg5gHfX8\n3+tdnZLFtsANufc3pmkjY6qqaX5aP5+30O6n6YMf/OCMrCqbTUen/6V+ra/TNsrMU7U6XQ01lHaR\nWbNgk03ezqxZc1m16po+r735FVp3cLvp/drWMLdrNo7K/D/1a51l/l/r97+rulwCJmkBMBkRB6X3\nxwIPR8TxuXnqEayZ2YiJiJ4yT52SxfrA74AXADcDFwKvjIgrKw3MzMzq0wwVEQ9KeivwQ2AW8BUn\nCjOzeqhNzcLMzOqrNldDlbkhT9Jn0+eXSprfzbIVxLRc0mWSlkm6cFgxSdpF0i8k/VXSO7v9PhXF\nVVVZvSrtt8sk/UzS7mWXrSimqsrp0BTTMkm/lrR/2WUrimkg5VQmrtx8+0h6UNLh3S475JjKl1VE\nVP5D1ux0DbATsAFwCfDUlnkOBr6XXj8D+GXZZYcdU3r/B2CLCsppK2Bv4MPAO7tZtoq4Ki6rZwJz\n0uuDavI3NWVMFZfTJrnXu5HdD1V1OU0Z06DKqZvvm+b7EXA2cHjVZdUupm7Lqi41izI35C0EFgNE\nxK+AuZK2LrnsMGOal/u839e9FcYUEbdFxMXA6m6XrSiupirK6hcRcXd6+ytgu7LLVhBTUxXldG/u\n7abA7WWXrSCmpkFcc1r2+/4T8C3gtmksO8yYmkqVVV2SRZkb8trN87gSyw47JsgulD5X0sWS3tSH\neMrGNIhlB73uOpTV0cD3prnsMGKCCstJ0mGSrgS+D7ytm2WHHBMMppxKxSVpW7KD9RdzsZRatoKY\nmq9LlVVdroYq28s+zDtUeo3p2RFxs6StgCWSroqIC4YUU7+XHfS6nxURt1RVVpKeD7wBeFa3y3ap\nl5igwnKKiDOBMyU9BzhZ0i49brfvMQFPSR8NopzKxvVpYFFEhCSx9vhQ5d9Uu5igi7KqS7K4Cdg+\n9357sgzZaZ7t0jwblFh2mDHdBBARN6fft0n6X7LqYq9/sGViGsSyA113RNySfg+9rFIH8peBgyLi\nzm6WHXJMlZZTLoYLlN0TtUWar7Jyao1J0mMi4o4BlVPZuPYCTsmOyWwJvEjS6pLLDjWmiPhOV2XV\nr86fHjtp1geuJeuk2ZDizuQFrO2MLFy2gphmA5ul15sAPwMOHEZMuXkneWQH90DKqQ9xVVZWwA5k\nnYMLpvt9hhhTleX0RNZeZr8ncG0NyqldTAMpp+l8X+Ak4GVVl1WHmLoqq54LsF8/wIvI7uC+Bjg2\nTft74O9z83w+fX4psGenZauMCXhC2mmXAL8ZZkzA1mRtmHcDdwLXA5sOspx6iavisjoRuANYln4u\nrPpvql1MFZfTe9I2l5Gdde5Tg3KaMqZBllOZuFrmXXNgrrKs2sXUbVn5pjwzMytUl6uhzMysxpws\nzMyskJOFmZkVcrIwM7NCThZmZlbIycLMzAo5WdhQSDpO0m9yw0rvW3VMvZD0VUnXpe+yNA050Y/1\nTqplCPfcZ6+VdHkaUnppu/l63P77+r1OGw9OFjZwkp4JvBiYHxF7kD0694bOS9VeAO+KiPnAO4Av\n9HG965D0IuCfgQMiYneyEQPunmreHh07gHXaGHCysGHYGrg9siGUiYg/RRqTRtJekhpp1MsfpGHn\nm9MvlXSJpBMkXZ6mv07S55orlnS2pOel1wdK+nl6GM5pkjZJ05enM/Zfp7Pyp6Tpm0o6KU27VNLL\nOq1nCs0B2X5JNvwEkmaleC9M6zwmt61zczEszH2H4yT9TtIFrB0Mr9WxZMOkrEhl+EBEnJiWf7qk\nX6btnSFpbprekLRXer2lpD/kyvAMSd+XdLWk49P0jwIbp9rSyZJmS/pu2geXSzqieFfbuHKysGE4\nB9g+HRD/Q9JzASRtAHyO7GEse5MNRfBvaZmTgLdExNPJzrbbDTUQQEjaEjgOeEFE7AX8muyMvznP\nbWn6F4F3pen/CtwZEbunGs+PCtbTzkFkwyVANqz4XRGxL9mgbG+StBOwCnhpWuf+wCdSGewFHAns\nQTbW2D5tvuvTUixT+Rrw7vQdLgc+kC+bNsvsARxB9uCgIyVtGxGLgFURMT8iXkM2jMRNEfH0iNgN\n+EFBOdgYq8uoszbGIuLedFB8DvB84FRJi8gOfk8jG08fsqd53SxpDtnT4n6aVnEy2YGrHZE1y+wK\n/Dyta0Pg57l5zki/lwIvS69fQHagbsZ5l6RDCtaT3+YJkj4C7MjaocQPBHaT9PL0fnPgSWQjgf57\n6tt4GHicsgdlPQc4IyL+CvxV0nfoYij+XFk1RwpdDJxeYtHzIuKetI4r0ne4qWWey4CPpxrH2bn9\nYTOQk4UNRUQ8DPwY+HFqUjqKLFn8NiL2y8/bbEbJT8q9fpBH1og3yr1eEhF/1yaE+9Pvh3jk3/1U\nB+ZO62lq9lmcIemtwPtZ+4Syt0bEkkd8Ael1ZMND7xkRD6UmoY3SevIxtEsUvyV7LO35BXG1K6uN\nWua7P/e6tUwAiIjfK3uu/IuBD0s6LyI+VLB9G1NuhrKBk7SzpCfnJs0HlpONlLmVpAVpvg0k7RoR\ndwF3SWqerb8qt+xy4OnKbE/W1BNk/QbPktTsO9ikZZtTWQK8JRfn3C7XI4CI+DxZM9szgR8Cb1b2\nzIfmd59NVsO4NSWK55OdyQfwE+AwSRtJ2gw4hKmbjv6drCYzL613Q0lHR/YI1jslPTvN9xqgkSur\nvdPrl1PO6lzs2wB/jYhvAB8nGwrcZijXLGwYNgU+lw7GDwK/B46JiNWpueazqTllfeBTwBXA64H/\nlhRkfR4ARMRP01n5FcCVpHb8iLg9nb1/U9Kj0uzHpW3l5dvxPwz8R6rpPARMRsSZJdfTXFfTh8lq\nFweTPVtgqbJ2rFuBw4BvAGdJugy4OMVORCyTdCrZEPe3AhdOVYAR8f2UKM5N6w3gK+njo4AvpaR0\nbSo7yA7wp6VO9u/m4u3Ul/FfwGWSfk3W/HeCpIeBB4B/bLOMzQAeotxqT9KOZG3mu1Udi9lM5WYo\nGwXNM2kzq4hrFmZmVsg1CzMzK+RkYWZmhZwszMyskJOFmZkVcrIwM7NCThZmZlbo/wM7iROrZbIs\n0wAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b05c8033490>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.hist(rhdList, bins=100)\n",
    "plt.title(\"RHD Locus Read Counts\")\n",
    "plt.xlabel(\"Sequence Read Counts\")\n",
    "plt.ylabel(\"Frequency\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
